usersimcrs/simulator/moviebot/moviebot_rl_sim.py

```diff
@@ -0,0 +1,422 @@
+"""User simulator to interact with MovieBot agent during reinforcement
+learning."""
+
+import logging
+import random
+from typing import Any, Dict, List, Tuple
+
+from nltk.stem import WordNetLemmatizer
+
+from dialoguekit.core.annotated_utterance import AnnotatedUtterance
+from dialoguekit.core.annotation import Annotation
+from dialoguekit.core.intent import Intent
+from dialoguekit.core.utterance import Utterance
+from dialoguekit.nlg import ConditionalNLG
+from dialoguekit.nlu.nlu import NLU
+from usersimcrs.domain.simulation_domain import SimulationDomain
+from usersimcrs.items.item_collection import ItemCollection
+from usersimcrs.items.ratings import Ratings
+from usersimcrs.simulator.agenda_based.interaction_model import (
+    InteractionModel,
+)
+from usersimcrs.simulator.user_simulator import UserSimulator
+from usersimcrs.user_modeling.simple_preference_model import PreferenceModel
+
+_LEMMATIZER = WordNetLemmatizer()
+
+
+class UserSimulatorMovieBot(UserSimulator):
+    def __init__(
+        self,
+        id: str,
+        preference_model: PreferenceModel,
+        interaction_model: InteractionModel,
+        nlu: NLU,
+        nlg: ConditionalNLG,
+        domain: SimulationDomain,
+        item_collection: ItemCollection,
+        ratings: Ratings,
+    ) -> None:
+        """Initializes the user simulator.
+
+        Args:
+            id: User simulator id.
+            preference_model: Preference model.
+            interaction_model: Interaction model.
+            nlu: Natural language understanding model.
+            nlg: Natural language generation model.
+            domain: Domain.
+            item_collection: Item collection.
+            ratings: Ratings.
+        """
+        super().__init__(id=id)
+        self._preference_model = preference_model
+        self._interaction_model = interaction_model
+        self._nlu = nlu
+        self._nlg = nlg
+        self._domain = domain
+        self._item_collection = item_collection
+        self._ratings = ratings
+        self.initialize_goal()
+
+    def _get_information_need(self) -> Dict[str, str]:
+        """Gets the information need for the goal.
+
```

Comment:
Can you explain the purpose of this method (similarly how it's done for `initialize_goal`)?

Reply:
Done.